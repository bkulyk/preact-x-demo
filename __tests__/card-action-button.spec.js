import React from 'react';
import CardActionButton from '@bitchin/card-action-button';
import { shallow } from './enzyme';

it('renders correctly, it is pretty much just a button', () => {
  expect(shallow(<CardActionButton>View</CardActionButton>))
    .toMatchSnapshot();
});

it('should send click events to the button', () => {
  expect(shallow(<CardActionButton onClick={() => {}}>View</CardActionButton>))
    .toMatchSnapshot();
});

it('should pass on any props to the button', () => {
  expect(shallow((
    <CardActionButton className="something" data-x="x" raised>View</CardActionButton>
  ))).toMatchSnapshot();
});
