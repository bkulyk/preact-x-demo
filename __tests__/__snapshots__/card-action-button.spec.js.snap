// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly, it is pretty much just a button 1`] = `
<Button
  className="mdc-card__action--button"
  raised={false}
>
  View
</Button>
`;

exports[`should pass on any props to the button 1`] = `
<Button
  className="mdc-card__action--button something"
  data-x="x"
  raised={true}
>
  View
</Button>
`;

exports[`should send click events to the button 1`] = `
<Button
  className="mdc-card__action--button"
  onClick={[Function]}
  raised={false}
>
  View
</Button>
`;
